piptools/_compat/click_compat.py

```diff
@@ -0,0 +1,5 @@
+import click
+from pip._vendor.packaging.version import Version
```

Comment:
I don't really like relying on how and whether pip vendors its deps. Plus it may cause problems when repackaged in distros that unbundle vendored software.

Reply:
sure, was just in analogy to `pip_compat` module - would you agree with using `int(click.__version__.split(".")[0])`?

Reply:
Yeah, that should be enough.

Reply:
Although, you may want to multi-line it to keep the inline complexity low

Reply:
not sure what you mean with multi-lining, as for me black would merge this into one line if not forcing to not do on the following: 
```
int(
    click.__version__.split(".")[0]
)
```

Reply:
nvmd, will simply add a comment then in between the brackeets

Reply:
Yeah, that's one of the reasons I dislike black...

As for the line complexity, I like this metric that https://wemake-python-stylegui.de introduced called Jones complexity. It really makes a difference maintainability/readability-wise!

Here's some more info on this https://sobolevn.me/2019/10/complexity-waterfall#lines.

I wish this set of linter plugins was integrated here.

Reply:
didn't knew the blog entry regarding the complexity until now, found it a good read and would recommend to colleagues, thanks a lot for that, very interesting!